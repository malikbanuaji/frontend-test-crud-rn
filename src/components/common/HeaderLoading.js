import React from 'react';
import {StyleSheet, Text, View, ActivityIndicator} from 'react-native';
import {Colors, Spacing, Typography} from '../../styles';

export default function HeaderLoading({text}) {
  return (
    <View style={styles.headerLoading}>
      <View style={styles.content}>
        <ActivityIndicator
          color={Colors.primary}
          animating={true}
          size={30}
          style={{marginRight: Spacing.smaller}}
        />
        {text && <Text style={styles.text}>{text}</Text>}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  headerLoading: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  content: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    ...Typography.headerLoadingText,
  },
});
