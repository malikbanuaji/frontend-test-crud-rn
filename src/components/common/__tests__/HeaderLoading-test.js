import React from 'react';
import renderer from 'react-test-renderer';
import HeaderLoading from '../HeaderLoading';

test('renders correctly', () => {
  const tree = renderer.create(<HeaderLoading />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with props `text`', () => {
  const tree = renderer.create(<HeaderLoading text={'Menunggu'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
