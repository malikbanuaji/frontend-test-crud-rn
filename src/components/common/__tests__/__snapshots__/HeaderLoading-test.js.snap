// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#93b5e1"
      hidesWhenStopped={true}
      size={30}
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly with props \`text\` 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#93b5e1"
      hidesWhenStopped={true}
      size={30}
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Medium",
          "fontSize": 14,
        }
      }
    >
      Menunggu
    </Text>
  </View>
</View>
`;
